security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package sds
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	authapi "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	sds "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/gogo/protobuf/types"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+var (
+	socket               string
+	fakeCertificateChain = []byte{01}
+	fakePrivateKey       = []byte{02}
+)
+
+func TestStreamSecrets(t *testing.T) {
+	socket = fmt.Sprintf("/tmp/gotest%q.sock", string(uuid.NewUUID()))
+	testHelper(t, socket, sdsRequestStream)
+}
+
+func TestFetchSecrets(t *testing.T) {
+	socket = fmt.Sprintf("/tmp/gotest%q.sock", string(uuid.NewUUID()))
```

Comment:
Have you checked whether a file will be created on the host and should we delete them as something like `defer func cleanup() {})`.

Reply:
from my local env, it looks like the socket file is automatically deleted after calling grpcListener.Close()

Reply:
Nit: is it better to use %s instead of %q?
Otherwise the file name will include `"`, e.g. `/tmp/gotest"<uuid>".sock` which looks weird.

Reply:
done